### PR TITLE
Application framework library with usage example

### PR DESCRIPTION
**Pull request by jaykumarark**

### CODE REVIEW

File: src/app-framework/example/main.cpp [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/**
+Apr 8, 2018
+Jay R Ravi
+**/
+
+#include <stdio.h>
```
Comment (jaykumarark) [inline src/app-framework/example/main.cpp]: Start here, this shows how to setup an example app by extending the physika::Application class

File: src/app-framework/include/app-framework/input.h [hunk 2]
```diff
@@ -0,0 +1,124 @@
+#pragma once
+
```
Comment (jaykumarark) [inline src/app-framework/include/app-framework/input.h]: This file contains physika's event code equivalent for keyboard and mouse events. Mapping from OS specific to physika codes are handled within each `Application` implementation (in this case ApplicationWin32 maps all windows event codes to physika codes).

File: src/app-framework/input.h [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    kMouseNone = 0, 
+    kMouseLeft = 1 << 1,
+    kMouseRight = 1 << 2,
+    kMouseMiddle = 1 << 3,
+    kMouseExtra1 = 1 << 4,
+    kMouseExtra2 = 1 << 5,
+};
+
+enum Keycode {
+    //! Alphabets
```
Comment (varunk08) [inline src/app-framework/input.h]: Strange comment convention. We should probably adopt some coding convention like google's for uniformity.

File: src/app-framework/input.h [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    kRightAlt, 
+    kSpace,
+    kEsc,
+
+    kKeyCodeCount, 
+};
+
+
+//! Ref: https://stackoverflow.com/questions/15889414/how-to-overload-operator-on-scoped-enum
+inline MouseButton operator|(MouseButton const& leftSide, MouseButton const& rightSide)
```
Comment (varunk08) [inline src/app-framework/input.h]: I would prefer const MouseButton&
Some one has better explained why that is preferable: https://stackoverflow.com/a/3695979/10487262
Also that is what we follow at work too.

File: src/app-framework/application-win32.cpp [hunk 5]
```diff
@@ -1,15 +1,264 @@
...
+    }
+    if (wParam & MK_MBUTTON) {
+        result |= kMouseMiddle;
+    }
+    return result; 
+}
+
+Keycode MapKeyWin32ToPhi(WPARAM wParam)
+{
+    switch (wParam) {
```
Comment (varunk08) [inline src/app-framework/application-win32.cpp]: A lookup table with all these mappings might be better for performance. It seems some compilers will optimize this away to a lookup table, but I am not sure of the details.

File: src/app-framework/application-win32.cpp [hunk 6]
```diff
@@ -1,15 +1,264 @@
 
+#include <stdio.h>
 #include <windows.h>
-
 #include "application-win32.h"
 
 namespace {
+using namespace physika;
 
 physika::ApplicationWin32* sApp = nullptr;
```
Comment (varunk08) [inline src/app-framework/application-win32.cpp]: I am not sure why we need this global ptr. It seems to be used mainly to call input handling functions below. Can't we just call the object's member functions directly?

File: CMakeLists.txt [hunk 7]
```diff
@@ -11,7 +11,7 @@ set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 
 # 3rd Party
 find_package(gtest)
-find_package(glfw)
+#find_package(glfw)
```
Comment (varunk08) [inline CMakeLists.txt]: Nitpick: remove line if not needed.

File: src/app-framework/application-win32.cpp [hunk 8]
```diff
@@ -344,35 +344,35 @@ void ApplicationWin32::Run()
     }
 }
 
-void physika::ApplicationWin32::OnUpdate()
+void ApplicationWin32::OnUpdate()
 {
 }
 
-void physika::ApplicationWin32::OnResize(int width, int height)
+void ApplicationWin32::OnResize(int /*width*/, int /*height*/ )
```
Comment (varunk08) [inline src/app-framework/application-win32.cpp]: Machi.. why da. My architect will kill me if I use this style for arguments.

Comment (jaykumarark) [reply src/app-framework/application-win32.cpp]: Arguments or comment style? What do you suggest?

File: CMakeLists.txt (hunk 7 above)
Comment (jaykumarark) [reply CMakeLists.txt]: Don't worry, I would have done the same nitpick comment. :) Keep the nitpicks coming. I'll it.

File: src/app-framework/application-win32.cpp (hunk 5 above)
Comment (jaykumarark) [reply src/app-framework/application-win32.cpp]: Interesting, I got to look that up ;)

**Pull request merged**